keysmash.c

```diff
@@ -1,17 +1,25 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <assert.h>
 
-char* keysmash(int length) {
-    char keys[] = { 'a', 's', 'd', 'f', 'j', 'k', 'l', 'a', 's', 'd', 'f', 'j', 'k', 'l', ';' };
-    length -= 2;
-    char* smash = (char*)malloc(sizeof(char)*1000);
+char* keysmash(unsigned length) {
+    char keys[16] = { 'a', 's', 'd', 'f', 'g', 'j', 'k', 'l', 'a', 's', 'd', 'f', 'j', 'k', 'l', ';' };
```

Comment:
@mxmeinhold To make it's length 16 so I can use `&15` and which should be faster than a modulo of 15

Reply:
I would call this premature optimisation. That's going to be cpu arch dependent, and the implementation in asm is going to be compiler dependent. There's no theoretical way to compare them. You'd have to set up in depth profiling of the asm in the runtime container.

Reply:
Hm. I can't think of a situation where modulo would be quicker than a primitive bitwise operation... The only way I can think of them being equal is with some magical instruction pipeline stuff...

Reply:
It's kinda hard to say what's fast in x86, since it's going to differ per cpu microcode, since the cpus aren't actually running straight x86 instructions, which is why you need profiling. I would optimise for readability in the vast majority of situations if you aren't doing that profiling.